packages/terra-status-view/src/StatusView.jsx

```diff
@@ -0,0 +1,211 @@
+        paddingTop = remainingHeight * 0.4;
+        paddingBottom = remainingHeight * 0.6;
+      }
+    } else if (!this.props.isAlignedTop) {
+      // glyph does not exist or does not fit, distribute remaining padding if any to center
+      const remainingHeight = viewHeight - componentsHeight;
+      if (remainingHeight > 0) {
+        paddingTop = remainingHeight * 0.4;
+        paddingBottom = remainingHeight * 0.6;
+      }
+    }
+
+    // set which components can fit into the view and add the additional 20px
+    // top and bottom padding to the remaining additional padding leftover
+    this.setState({
+      showGlyph,
+      paddingTop,
+      paddingBottom,
+    });
+  }
+
+  render() {
+    const {
+    children,
+    isAlignedTop,
+    isGlyphHidden,
+    message,
+    title,
+    variant,
+    ...customProps
+    } = this.props;
+
+    const { intl } = this.context;
+
+    let glyphSection;
+    if (!isGlyphHidden && this.state.showGlyph) {
+      glyphSection = <div className={cx('glyph')} ref={(element) => { this.glyphNode = element; }}><svg className={cx(variant)} /></div>;
+    }
+    let messageSection;
+    if (message) {
+      messageSection = <div className={cx('message')} ref={(element) => { this.messageNode = element; }}>{message}</div>;
+    }
+    let actionSection;
+    if (children.length) {
+      actionSection = <div className={cx('actions')} ref={(element) => { this.actionsNode = element; }}>{children}</div>;
+    }
+    let defaultTitle = title;
+    if (!defaultTitle) {
+      defaultTitle = (variant === StatusView.variants.CUSTOM) ? '' : intl.formatMessage({ id: `Terra.status-view.${variant}` });
+    }
+    let dividerSection;
+    if (messageSection || actionSection) {
+      dividerSection = <div className={cx('divider')} ref={(element) => { this.dividerNode = element; }}><Divider /></div>;
+    }
+
+    const titleSection = <div className={cx('title')} ref={(element) => { this.titleNode = element; }}>{defaultTitle}</div>;
+
+    const statusViewClassNames = cx([
+      'status-view',
+      customProps.className,
+    ]);
+
+    const statusViewInnerStyles = {
+      paddingTop: `${this.state.paddingTop}px`,
+      paddingBottom: `${this.state.paddingBottom}px`,
+    };
+
+    return (
+      <div
+        {...customProps}
+        className={statusViewClassNames}
+      >
+        <div
+          className={cx('status-view-inner')}
+          style={{ ...statusViewInnerStyles }}
+          ref={(element) => { this.innerNode = element; }}
+        >
+          {glyphSection}
+          <div className={cx('message-content-group')} >
+            {titleSection}
+            {dividerSection}
+            {messageSection}
+          </div>
+          {actionSection}
+        </div>
+      </div>
+    );
+  }
+}
+
+StatusView.propTypes = propTypes;
+StatusView.contextTypes = contextTypes;
+StatusView.defaultProps = defaultProps;
+StatusView.variants = variants;
```

Comment:
This should probably be StatusView.Opts.variants to align with other components.

Reply:
done